src/Parallel/Main.hpp

```diff
@@ -671,6 +745,17 @@ void contribute_to_phase_change_reduction(
         "reduction.");
   }
 }
+
+template <typename Metavariables>
+std::string Main<Metavariables>::current_checkpoint_dir() const noexcept {
```

Comment:
[random line]: since we are preventing duplicate checkpoints (a good thing!) should we have something that gets run very early on in the `Main` constructor that verifies that no `checkpoint` directories exist? I'd be rather unhappy if I'm doing a multi-hour run and when it checkpoints it tells me it can't and I lose that time :P This check should also be run as part of our `--check-options` (or whatever exactly it is) flag

Reply:
On line 367, Main is checking that `SpectreCheckpoint0` (with more 0's after will's review) does not exist. Are you requesting I check that nothing named `SpectreCheckpoint*` exists?

Do I really want to run this check as part of our input-file parsing? To me checking for checkpoint dirs feels outside the scope of input file / option checking. Docs etc will need updating. I could add another command-line flag for this a new check?

Reply:
Ah, yes that does (almost) the full check. I think checking that no checkpoint files exist would be the right thing to do to avoid the case where someone just deletes `SpectreCheckpoint0000000`.

I personally think I do want to run this as part of the input file check since I've always viewed the input file check as "check that the simulation will succeed except for runtime physics failures". I would include writing checkpoints in that class of failures :shrug: In the future we may even want to specify the checkpoint file prefix as an input file option :shrug:

Reply:
Okay, fair enough 🤷... but would you agree with renaming `--check-options` to something more broad like `--check-inputs` that can be documented to encompass options + checkpoint state?

Reply:
yea, that sounds good to me :)

Reply:
I'm confused now.  If we require old checkpoints to be removed before running, why are we numbering them?  And don't we need one to do the restart from?

Reply:
@wthrowe My current plan is to check that,
- when first starting the exec or when running input file checks: no checkpoint files exist
- when restoring from a checkpoint: no _future_ checkpoint files exist
- just before writing a checkpoint file, that particular file doesn't exist. (this should be redundant with the two above, but should help provide readable errors instead of who-knows-what if anything goes wrong.)

What do you think of this proposal?

Reply:
Ah, OK.  I missed that the check being discussed was just for the start of the simulation, not for each restart.  That all makes sense then.